plugins/Terminal/js/reducers/commandline.js

```diff
@@ -0,0 +1,111 @@
+import { Map, List } from 'immutable'
+import * as constants from '../constants/commandline.js'
+
+const initialState = Map({
+	commandHistory: List([]),
+	currentCommand: '',
+	commandIndex: 0,
+	showWalletPrompt: false,
+	showSeedPrompt: false,
+	showCommandOverview: true,
+})
+
+export default function commandLineReducer(state = initialState, action) {
+	switch (action.type) {
+
+	case constants.ADD_COMMAND:
+			//Add command to the command history.
+		console.log('NEW COMMAND!')
+		var newCommandHistory = state.get('commandHistory').push(action.command)
+		var newState = state.set('commandHistory', newCommandHistory)
+		newState = newState.set('commandIndex', 0)
+		newState = newState.set('currentCommand', '')
+		return newState
+
+
+	case constants.UPDATE_COMMAND:
+			//Updates output of command given by command name and id.
+		var newCommandHistory = state.get('commandHistory')
+		var commandArray = newCommandHistory.findLastEntry(
+				(val) => (val.get('command') === action.command && val.get('id') === action.id)
+			)
+
+		if (!commandArray) {
+			console.log(`Error did not find command: { command: ${action.command}, id: ${action.id} } in command history: ${JSON.stringify(newCommandHistory)}`)
+			return state
+		}
+
+		var [commandIdx, newCommand] = commandArray
+
+		if (action.dataChunk) {
+			newCommand = newCommand.set('result', newCommand.get('result') + action.dataChunk)
+		}
+
+		if (action.stat) {
+			newCommand = newCommand.set('stat', action.stat)
+		}
+
+		console.log(newCommand)
+		newCommandHistory = newCommandHistory.set(commandIdx, newCommand)
+		return state.set('commandHistory', newCommandHistory)
```

Comment:
Which side effects are you referring to, exactly? The console logging definitely doesn't belong, but the rest seems pure.